### PR TITLE
feat: new `github-teams` script

### PR DESCRIPTION
**Pull request by jzempel**
## Description

This PR adds a new `garden github-teams` CLI, corresponding `githubTeams` API, and supporting documentation. See the new README for details.

## Detail

To be used along with GH Actions conditionals for job security :wink:

### CODE REVIEW

File: src/github/teams/index.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ *
+ * @returns {Promise<string[]>} A list of GitHub teams.
+ */
+export const execute = async (args: IGitHubTeamsArgs): Promise<string[] | undefined> => {
+  let retVal: string[] | undefined;
+
+  try {
+    const auth = args.token || (await getToken(args.spinner));
+    const github = new Octokit({ auth });
+    const org = (args.org || (await getRepository(args.path, args.spinner))?.owner)!;
```
Comment (ze-flo) [inline src/github/teams/index.ts:40]: `getRepository` handles a missing path. ✅

**Review by ze-flo — APPROVED** [review]
Fantastic work! 💯 🔥 🚀

**Review by geotrev — CHANGES_REQUESTED** [review]
Just the tiny typo. Otherwise :shipit:

File: src/github/teams/index.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  return retVal;
+};
+
+export default (spinner: Ora): Command => {
+  const command = new Command('github-teams');
+
+  return command
+    .description('output GitHub organization teams')
+    .argument('[org]', 'GitHub organization name; defaults to repository owner')
+    .option('-u --user [user]', 'user to get team membership for; defaults to current')
```
[the remark this answers is not in the document]
Comment (jzempel) [reply src/github/teams/index.ts]: Nice catch 😅

**Review by geotrev — APPROVED** [review]

**Pull request merged**